reference/tools.rst

```diff
@@ -1659,6 +1659,7 @@ It raises a ``ConanInvalidConfiguration`` when is not supported.
 * If the current cppstd does not support C++17, ``check_min_cppstd`` will raise an ``ConanInvalidConfiguration`` error.
 * If ``gnu_extensions`` is True, it is required that the applied ``cppstd`` supports the gnu extensions.
   (e.g. gnu17), otherwise, an :ref:`ConanInvalidConfiguration<conditional_settings_options_requirements>` will be raised. The ``gnu_extensions`` is checked in any OS.
+* If no compiler has been specified or the compiler is unknown, it raises a ``ConanInvalidConfiguration`` exception.
```

Comment:
I think it will raise a `ConanException`:

```python
        compiler = conanfile.settings.get_safe("compiler")
        compiler_version = conanfile.settings.get_safe("compiler.version")
        if not compiler or not compiler_version:
            raise ConanException("Could not obtain cppstd because there is no declared "
                                 "compiler in the 'settings' field of the recipe.")
```

If it is a `ConanException`, I'm not sure if we should document it

Reply:
That's right! Thanks Javi, I'll update it.